R/fix_dates.R

```diff
@@ -0,0 +1,140 @@
+months <- new.env()
+months$abbrev <- tolower(month.abb)
+months$full <- tolower(month.name)
+
+#' @title Clean up messy date columns
+#' @description Cleans up a \code{dataframe} object which has date columns
+#' entered via a free-text box (possibly by different users) and are therefore
+#' in a non-standardised format. Supports numerous seperators including /,- or
+#' space. Supports all-numeric, abbreviation or long-hand month notation. Where
+#' day of the month has not been supplied, the first day of the month is
+#' imputed. When day, month and year is given either DMY or YMD is assumed; the
+#' US system of MDY is not supported.
+#' @param df A \code{dataframe} object with messy date column(s)
+#' @param col.names Character vector of names of columns of messy date data
+#' @return A \code{dataframe} object. Selected columns are of type \code{Date}
+#' @examples
+#' bad.dates <- data.frame(id = seq(5),
+#'                         some.dates = c("02/05/92",
+#'                                        "01-04-2020",
+#'                                        "1996/05/01",
+#'                                        "2020-05-01",
+#'                                        "02-04-96"),
+#'                         some.more.dates = c("2015",
+#'                                             "02/05/00",
+#'                                             "05/1990",
+#'                                             "2012-08",
+#'                                             "jan 2020"))
+#'fixed.df <- fix_dates(bad.dates, c("some.dates", "some.more.dates"))
+#' @export
+fix_dates <- function(df, col.names) {
+  if (is.data.frame(df) == F) {
```

Comment:
Should be `if (!is.data.frame(df))`, not wise to use `F` because it's not a reserved name